apollo-testing-support/src/commonMain/kotlin/com/apollographql/apollo3/testing/TestNetworkTransport.kt

```diff
@@ -0,0 +1,86 @@
+package com.apollographql.apollo3.testing
+
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.api.ApolloRequest
+import com.apollographql.apollo3.api.ApolloResponse
+import com.apollographql.apollo3.api.Error
+import com.apollographql.apollo3.api.Operation
+import com.apollographql.apollo3.network.NetworkTransport
+import com.benasher44.uuid.uuid4
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+
+class TestNetworkTransport(
+    val handler: TestNetworkTransportHandler = MapTestNetworkTransportHandler(),
+) : NetworkTransport {
+  override fun <D : Operation.Data> execute(request: ApolloRequest<D>): Flow<ApolloResponse<D>> {
+    @Suppress("UNCHECKED_CAST")
+    return flowOf(handler.handle(request) as ApolloResponse<D>)
+  }
+
+  fun <D : Operation.Data> register(
+      operation: Operation<D>,
+      response: ApolloResponse<D>,
+  ) = (handler as MapTestNetworkTransportHandler).register(operation, response)
+
+  fun <D : Operation.Data> register(
+      operation: Operation<D>,
+      data: D? = null,
+      errors: List<Error>? = null,
+  ) = (handler as MapTestNetworkTransportHandler).register(operation, data, errors)
+
+  override fun dispose() {}
+}
+
+interface TestNetworkTransportHandler {
+  fun handle(request: ApolloRequest<*>): ApolloResponse<*>
+}
+
+class QueueTestNetworkTransportHandler : TestNetworkTransportHandler {
+  private val queue = ArrayDeque<ApolloResponse<out Operation.Data>>()
+
+  fun <D : Operation.Data> enqueue(response: ApolloResponse<D>) {
+    queue.add(response)
```

Comment:
I think this needs to be locked. On native it doesn't throw because we run the flow on the main thread always but it might not always be the case. And on the JVM we cannot make any assumption about what dispatcher is going to be used.